ha/monitor/k8s/object_monitor.py

```diff
@@ -64,20 +98,40 @@ def run(self):
         k8s_watch = watch.Watch()
 
         # Get method pointer
-        object_function = getattr(k8s_client, ObjectMap.get_subscriber_func(self._object))
-
-        # Start watching events corresponding to self._object
-        for an_event in k8s_watch.stream(object_function, **self._args):
-            Log.debug(f"Received event {an_event}")
-            alert = EventParser.parse(self._object, an_event, self._object_state)
-            if alert is None:
-                continue
-            if self._starting_up:
-                if an_event[K8SEventsConst.TYPE] == EventStates.ADDED:
-                    alert.is_status = True
-                else:
-                    self._starting_up = False
-
-            # Write to message bus
-            Log.info(f"Sending alert on message bus {alert.to_dict()}")
-            self._producer.publish(str(alert.to_dict()))
+        self._object_function = getattr(k8s_client, ObjectMap.get_subscriber_func(self._object))
+        Log.info(f'Starting watch on {self._object} events: {self._object_function.__name__}')
+
+        # While True loop to restart the watch.Watch.stream() after specified timeout
+        while True:
+            # create a object of Watch.stream generator loop on it.
+            k8s_watch_stream =  k8s_watch.stream(self._object_function, **self._args)
+            # Start watching events corresponding to self._object
+            for an_event in k8s_watch_stream:
+
+                # Check for the signals (SIGTERM signal)
+                self.check_for_signals(k8s_watch, k8s_watch_stream)
+
+                # Due to SIGTERM signal stopping further event processing
+                if self._stop_event_processing:
+                    Log.info(f"Stopped processing {self._object} events.")
+                    break
+
+                Log.debug(f"Received event {an_event}")
+                alert = EventParser.parse(self._object, an_event, self._object_state)
+                if alert is None:
+                    continue
+                if self._starting_up:
+                    if an_event[K8SEventsConst.TYPE] == EventStates.ADDED:
+                        alert.is_status = True
+                    else:
+                        self._starting_up = False
+
+                # Write to message bus
+                Log.info(f"{self._object}_monitor Sending alert on message bus {alert.to_dict()}")
+                self._producer.publish(str(alert.to_dict()))
+
+            # If stop processing events is set then no need to retry just break the loop and
+            # If we don't specify timeout no need to restart the loop it will happen internally
+            if self._stop_event_processing or K8SClientConst.TIMEOUT_SECONDS not in self._args:
```

Comment:
1. When would TIMEOUT_SECONDS not be present in _args  since we are always passing it ?
2. How will loop restart in this condition ?

Reply:
if tomorrow we don't need to pass timeout in some cases for some specific monitor thread.
in that case, while True is not required as the loop inside the Watch.stream will make sure to continue which will not be the case if we provide the timeout.
so in that case, if we don't provide timeout then no need to retry just break the loop as Watch.stream will be blocking the call which has while True loop so no need for a loop for it.